### PR TITLE
Store/load local to user ID and host to owned global IDs arrays

### PR DESCRIPTION
**Pull request by roshandathathri**

### CODE REVIEW

**Review by scober — APPROVED** [review]
```
  _   _   _   _
 / \ / \ / \ / \
( L | G | T | M )
 \_/ \_/ \_/ \_/
```

**Review by tylershunt — APPROVED** [review]

File: libgalois/include/katana/PropertyGraph.h [hunk 1]
```diff
@@ -232,12 +232,30 @@ class KATANA_EXPORT PropertyGraph {
...
   // TODO(witchel): ChunkedArray is inherited from arrow::Table interface but this is
-  // really a ChunkedArray of one change, change to arrow::Array.
+  // really a ChunkedArray of one chunk, change to arrow::Array.
+  const std::shared_ptr<arrow::ChunkedArray>& host_to_owned_global_ids() const {
+    return rdg_.host_to_owned_global_ids();
+  }
+  void set_host_to_owned_global_ids(std::shared_ptr<arrow::ChunkedArray>&& a) {
+    rdg_.set_host_to_owned_global_ids(std::move(a));
+  }
+
```
Comment (witchel) [inline libgalois/include/katana/PropertyGraph.h:242]: Do we want to continue with `arrow::ChunkedArray`?

File: libtsuba/src/RDG.cpp [hunk 2]
```diff
@@ -172,13 +174,17 @@ tsuba::RDG::AddPartitionMetadataArray(
     AddMirrorNodes(std::move(col));
   } else if (name.find(kMasterNodesPropName) == 0) {
     AddMasterNodes(std::move(col));
+  } else if (name == kHostToOwnedGlobalIDsPropName) {
+    set_host_to_owned_global_ids(std::move(col));
+  } else if (name == kLocalToUserIDPropName) {
```
Comment (witchel) [inline libtsuba/src/RDG.cpp:179]: Is this where we pay for sparsity?  If globalIDs were dense, we'd only need 2 values to represent the interval.

**Review by witchel — APPROVED** [review]
Looks good, though if you are adding these keys (like kHostToOwnedGlobalIDsPropName), don't the `to_json` and` from_json` need editing?

File: libtsuba/src/RDG.cpp (hunk 2 above)
Comment (roshandathathri) [reply libtsuba/src/RDG.cpp:179]: This is an interval per host. It is assumed to be dense but that's not a strict requirement.

File: libgalois/include/katana/PropertyGraph.h (hunk 1 above)
Comment (roshandathathri) [reply libgalois/include/katana/PropertyGraph.h:242]: Yes, I lean towards supporting them.

**Comment by roshandathathri** [comment]
> if you are adding these keys (like kHostToOwnedGlobalIDsPropName), don't the `to_json` and` from_json` need editing?

@tylershunt I thought this wasn't required.

**Comment by tylershunt** [comment]
> > if you are adding these keys (like kHostToOwnedGlobalIDsPropName), don't the `to_json` and` from_json` need editing?
> 
> @tylershunt I thought this wasn't required.

Shouldn't be required in this case. These values are treated like pretty much like properties but they have well defined names that tsuba uses to load them. It should be added to this list, then it'll be written out as part of the `PartHeader` JSON object here:
https://github.com/KatanaGraph/katana/blob/ecf147d4fd2cb6b3666fc2286395ce4f06e524bb/libtsuba/src/RDGPartHeader.cpp#L396

**Comment by roshandathathri** [comment]
Thanks, Tyler!

File: libtsuba/src/RDG.cpp (hunk 2 above)
Comment (witchel) [reply libtsuba/src/RDG.cpp:179]: It looks like we are storing the entire ChunkedArray.  If it were dense, we would only need to store the top and bottom values.

Comment (roshandathathri) [reply libtsuba/src/RDG.cpp:179]: I'm confused. `host_to_owned_global_ids` is O(H) array where H is the number of hosts/partitions.

Comment (witchel) [reply libtsuba/src/RDG.cpp:179]: Ooops!  My bad.  I saw global ids and thought it was all of them.  Sorry!

**Pull request merged**